src/attributes/compiler.jl

```diff
@@ -0,0 +1,272 @@
+abstract type CompilerAttribute <: MOI.AbstractOptimizerAttribute end
+
+@doc raw"""
+    ARCHITECTURE()
+""" struct ARCHITECTURE <: CompilerAttribute end
```

Comment:
types should no be all caps

Reply:
JuMP standard is TimeLimitSec ....

Reply:
I begun by using all caps because we were already using the very similar names for flags/attrs as there were internal types i.e. `Encoding` and `AbstractArchitecture`. I wanted to avoid confusion between the two. This also distinguishes ToQUBO's from other solver/MOI attrs

Reply:
Let's follow @joaquimg advice here. You can add a prefix for the things only relevant to this repo to differentiate it from others.

Reply:
So, to avoid naming clashes I suggest that we write this submodule `Flags` or `Settings` or something to hold such attributes. It would be used as
```
using JuMP
using ToQUBO

model = Model(ToQUBO.Optimizer)

set_optimizer_attribute(model, ToQUBO.Flags.DefaultVariableEncodingBits(), 5)

optimize!(model)
```
or even

```
using JuMP
using ToQUBO
using ToQUBO.Flags

model = Model(ToQUBO.Optimizer)

set_optimizer_attribute(model, DefaultVariableEncodingBits(), 5)

optimize!(model)
```